assignment7.txt

```diff
@@ -0,0 +1,24 @@
+Bettina Benitez
+
+1. ++*p reads right to left. Therefore, it increments point p.  Same with *++p, the order of operation is right to left. But point p is incremented then the dereferenced. *p++ is read left to right therefore it is incremented then dereferenced. 
+
+2. No the order of precedence is not guaranteed
+
+3. advantages:
+- it points to a specific lovation in memory
+- provides direct access to memory
+- reduces storage and complexity
+
+4
+ - 4.1: *char
+ - 4.2: invalid
+ - 4.3: invalid 
```

Comment:
This is perfectly valid! '\0' is just a NULL terminator and by definition, NULL is equal to 0. So this would evaluate as true, which is 1 in C.